src/videojs-hls.js

```diff
@@ -894,7 +925,11 @@ videojs.Hls.getPlaylistDuration = function(playlist, startIndex, endIndex) {
 
   for (; i >= startIndex; i--) {
     segment = playlist.segments[i];
-    dur += (segment.duration !== undefined ? segment.duration : playlist.targetDuration) || 0;
+    if (segment.preciseDuration) {
+      dur += segment.preciseDuration;
```

Comment:
More concisely (though maybe not more readably): `dur += segment.preciseDuration || segment.duration || playlist.targetDuration || 0`